onvm/onvm_nflib/onvm_nflib.c

```diff
@@ -405,34 +417,32 @@ onvm_nflib_run_callback(struct onvm_nf_context *nf_context, pkt_handler_func han
 void *
 onvm_nflib_thread_main_loop(void *arg) {
         struct rte_mbuf *pkts[PACKET_READ_SIZE];
-        struct onvm_nf_context *nf_context;
+        struct onvm_nf_local_ctx *nf_local_ctx;
         struct onvm_nf *nf;
         uint16_t nb_pkts_added;
-        struct onvm_nf_info *info;
         pkt_handler_func handler;
-        callback_handler_func callback;
+        callback_func callback;
         uint64_t start_time;
         int ret;
 
-        nf_context = (struct onvm_nf_context *)arg;
-        nf = nf_context->nf;
-        onvm_threading_core_affinitize(nf->info->core);
+        nf_local_ctx = (struct onvm_nf_local_ctx *)arg;
+        nf = nf_local_ctx->nf;
+        onvm_threading_core_affinitize(nf->thread_info.core);
 
-        info = nf->info;
-        handler = nf->nf_pkt_function;
-        callback = nf->nf_callback_function;
+        handler = nf->functions.pkt_handler;
```

Comment:
Storing the NF functions as variables here is unnecessary.  It would be better to say `nb_pkts_added = onvm_nflib_dequeue_packets((void **)pkts, nf, nf->functions.handler);` instead of `nb_pkts_added = onvm_nflib_dequeue_packets((void **)pkts, nf, handler);`.  The former is more informative compared to `handler`.

same comment for `callback` below.  i don't like using a variable for a value that is only used once--using the struct/ptr field is sufficient.

Reply:
Should be resolved now